tests/console/commands/test_add.py

```diff
@@ -1999,34 +2014,54 @@ def test_add_with_lock_old_installer(
 
 
 def test_add_keyboard_interrupt_restore_content(
-    app: PoetryTestApplication,
+    poetry_with_up_to_date_lockfile: Poetry,
     repo: TestRepository,
-    installer: NoopInstaller,
-    tester: CommandTester,
+    command_tester_factory: CommandTesterFactory,
     mocker: MockerFixture,
 ):
+    tester = command_tester_factory("add", poetry=poetry_with_up_to_date_lockfile)
+
     mocker.patch(
         "poetry.installation.installer.Installer.run", side_effect=KeyboardInterrupt()
     )
-    original_content = app.poetry.file.read()
+    original_pyproject_content = poetry_with_up_to_date_lockfile.file.read()
+    original_lockfile_content = poetry_with_up_to_date_lockfile._locker.lock_data
 
     repo.add_package(get_package("cachy", "0.2.0"))
+    repo.add_package(get_package("docker", "4.3.1"))
 
-    tester.execute("cachy --dry-run")
+    tester.execute("cachy")
```

Comment:
Updated the test here because since we're testing a keyboard interrupt, it may be nice to test that the non-dry-run variant prevents updates on the files.